src/native/corehost/deps_format.cpp

```diff
@@ -439,7 +439,7 @@ bool deps_json_t::has_package(const pal::string_t& name, const pal::string_t& ve
 bool deps_json_t::load(bool is_framework_dependent, const pal::string_t& deps_path, const rid_fallback_graph_t& rid_fallback_graph)
 {
     m_deps_file = deps_path;
-    m_file_exists = bundle::info_t::config_t::probe(deps_path) || pal::file_exists(deps_path);
+    m_file_exists = pal::realpath(&m_deps_file, true) || bundle::info_t::config_t::probe(m_deps_file);
```

Comment:
Can you double check that the single-file bundle probe works as expected? I think the bundle just uses `get_deps_from_app_binary` / `get_runtime_config_path` to build up the paths, so I don't think they compute/store the real resolved path?